process/process.go

```diff
@@ -17,41 +17,109 @@ import (
 
 type processesFn func() ([]ps.Process, error)
 
-func findPIDsWithFn(fn processesFn, prefix string, log logging.Logger) ([]int, error) {
-	processes, err := fn()
+type matchFn func(ps.Process) bool
+
+func matchPath(p ps.Process, match string, log logging.Logger) bool {
+	path, err := p.Path()
+	if err != nil {
+		log.Warningf("Unable to get path for process: %s", err)
+		return false
+	}
+	return path == match
+}
+
+func containsPath(p ps.Process, match string, log logging.Logger) bool {
+	path, err := p.Path()
+	if err != nil {
+		log.Warningf("Unable to get path for process: %s", err)
+		return false
+	}
+	return strings.Contains(path, match)
+}
+
+// FindProcesses returns processes containing string matching process path
+func FindProcesses(match string, wait time.Duration, delay time.Duration, log logging.Logger) ([]ps.Process, error) {
+	log.Infof("Finding process %q (max wait %s)", match, wait)
+	matchPath := func(p ps.Process) bool { return containsPath(p, match, log) }
+	start := time.Now()
+	for i := 0; time.Since(start) < wait || i == 0; i++ {
+		log.Debugf("Find process %q (%s < %s)", match, time.Since(start), wait)
+		procs, err := findProcessesWithFn(ps.Processes, matchPath, 0)
+		if err != nil {
+			return nil, err
+		}
+		if len(procs) > 0 {
+			return procs, nil
+		}
+		time.Sleep(delay)
+	}
+	return nil, nil
+}
+
+// findProcessWithPID with return process for a pid
+func findProcessWithPID(pid int) (ps.Process, error) {
+	matchPID := func(p ps.Process) bool { return p.Pid() == pid }
+	procs, err := findProcessesWithFn(ps.Processes, matchPID, 1)
+	if err != nil {
+		return nil, err
+	}
+	if len(procs) == 0 {
+		return nil, nil
+	}
+	return procs[0], nil
+}
+
+// Ignore deadcode warning
+var _ = findProcessWithPID
```

Comment:
So this is a useful function but is currently only used by tests... I don't want to nuke it though and don't want to move it into test files. Basically am ok with a deadcode ignore here.... I made the comment explain this more.